apps/heft/src/cli/actions/BuildAction.ts

```diff
@@ -108,4 +240,25 @@ export class BuildAction extends HeftActionBase<IBuildActionData, BuildHooks> {
       watchMode: this._watchFlag.value
     };
   }
+
+  protected async _runPhaseWithLogging(buildPhaseName: string, buildPhase: IBuildPhase): Promise<void> {
+    if (buildPhase.hooks.run.isUsed()) {
+      await this._runWithLogging(buildPhaseName, async () => await buildPhase.hooks.run.promise());
+    }
+  }
+
+  private async _runWithLogging(buildPhaseName: string, fn: () => Promise<void>): Promise<void> {
+    this.terminal.writeLine(` ---- ${buildPhaseName} started ---- `);
```

Comment:
Can phases run in parallel? If so, should we be using `stream-collator`?

Reply:
Phases are serial in non-`--watch` mode. I've been prefixing everything's logs with something like `[copy-static-assets] <log message>`, so it's easy to keep track of what's coming from where when things are running in parallel.